truffle/src/main/ruby/core/rubinius/common/array.rb

```diff
@@ -612,6 +612,29 @@ def flatten!(level=-1)
     nil
   end
 
+
+  def inspect
+    return "[]".force_encoding(Encoding::US_ASCII) if @total == 0
+    comma = ", "
+    result = "["
+
+    return "[...]" if Thread.detect_recursion self do
+      each_with_index do |element, index|
+        temp = element.inspect
+        result.force_encoding(temp.encoding) if index == 0
+        result << temp << comma
+      end
+    end
+
+    Rubinius::Type.infect(result, self)
+    #result.shorten!(2) # MODIFIED We don't implement string shorten.
```

Comment:
Did you make this change? We'd much prefer shimming `shorten!` to do nothing rather than have this line commented out.